example/lib/pages/post_detail/post_detail_page_scenes.dart

```diff
@@ -17,35 +26,26 @@ class PostDetailPageScene extends Scene {
   @override
   List<EnvironmentControlBuilder> get environmentControlBuilders =>
       <EnvironmentControlBuilder>[
-        (_, VoidCallback rebuildScene) {
+        (_, EnvironmentState state) {
           return DropdownControl<Post>(
-              value: currentPost,
+              value: state.get<Post>(key: _currentPostKey)!,
               title: const Text('Post'),
               items: Post.fakePosts(),
               onChanged: (Post? newPost) {
                 if (newPost == null) {
                   return;
                 }
 
-                // When a new [Post] is chosen, call [rebuildScene()] to update
-                // the UI with the newly chosen post.
-                currentPost = newPost;
-                rebuildScene();
+                state.set(key: _currentPostKey, value: newPost);
               });
         },
       ];
 
   @override
-  void onEnvironmentReset() {
-    super.onEnvironmentReset();
-    currentPost = Post.fakePosts().first;
-  }
-
-  @override
-  Widget build() {
+  Widget build(BuildContext context) {
     return EnvironmentAwareApp(
       home: PostDetailPage(
-        post: currentPost,
+        post: context.read<EnvironmentState>().get<Post>(key: _currentPostKey)!,
```

Comment:
This indexing is a bit of a mouthful. At first glance one questions I have is whether the "Post" type here is part of the key, similar to InheritedWidget / Guice etc?

Reply:
This just tells the compiler what the return type of `get` will be. The alternative to this would be `context.read<EnvironmentState>().get(key: _currentPostKey) as Post`.